.github/workflows/sycl_resolve_test_matrix.yml

```diff
@@ -0,0 +1,48 @@
+name: Reusable test matrix generation
+
+on:
+  workflow_call:
+    inputs:
+      intel_drivers_image:
+        type: string
+        required: false
+        default: "ghcr.io/intel/llvm/ubuntu2004_intel_drivers:latest"
+      amdgpu_image:
+        type: string
+        required: false
+        default: "ghcr.io/intel/llvm/ubuntu2004_build:latest"
+      cuda_image:
+        type: string
+        required: false
+        default: "ghcr.io/intel/llvm/ubuntu2004_build:latest"
+      lts_config:
+        type: string
+        required: true
+    outputs:
+      lts:
+        description: "Generated Matrix"
+        value: ${{ jobs.resolve_matrix.outputs.lts }}
+jobs:
+  resolve_matrix:
+    name: Resolve Test Matrix
+    runs-on: ubuntu-latest
+    outputs:
+      lts: ${{ steps.work.outputs.lts }}
+    steps:
+    - name: Download scripts and configs
+      shell: bash
+      run: |
+        wget raw.githubusercontent.com/intel/llvm/${{ github.sha }}/devops/scripts/generate_test_matrix.js
+        wget raw.githubusercontent.com/intel/llvm/${{ github.sha }}/devops/test_configs.json
+        wget raw.githubusercontent.com/intel/llvm/sycl/devops/dependencies.json
+        mv dependencies.json dependencies.sycl.json
+        wget raw.githubusercontent.com/intel/llvm/${{ github.sha }}/devops/dependencies.json
+    - id: work
+      uses: actions/github-script@v6
+      name: Generate matrix
+      env:
+        GHA_INPUTS: ${{ toJSON(inputs) }}
+      with:
+        script: |
+          const script = require('./generate_test_matrix.js');
```

Comment:
Yes, the only purpose of the script is to process some config files (test configs, deps), this is also usually done on github runner, rather than self-hosted since task is trivial. For Windows workflow such processing can be done on Linux machine as well.

Reply:
Sorry, I meant to write "for Windows". It seems to me that it process only Linux config files, but not Windows.